src/main/java/games/strategy/engine/framework/GameDataManager.java

```diff
@@ -106,8 +92,7 @@ static GameData loadGameInProxySerializationFormat(final InputStream is) throws
   }
 
   private static Memento loadMemento(final InputStream is) throws IOException {
-    try (final GZIPInputStream gzipis = new GZIPInputStream(is);
-        final ObjectInputStream ois = new ObjectInputStream(gzipis)) {
+    try (final ObjectInputStream ois = new ObjectInputStream(new GZIPInputStream(is))) {
```

Comment:
This is not safe.  If the `ObjectInputStream` constructor throws, the `GZIPInputStream` will not be closed.